sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -529,6 +533,133 @@ class Analyzer(
           makeGeneratorOutput(p.generator, p.generatorOutput), p.child)
     }
   }
+
+  object ResolveWindowFunction extends Rule[LogicalPlan] {
+    def hasWindowFunction(projectList: Seq[NamedExpression]): Boolean = {
+      projectList.foreach ( _.foreach {
+        case window: WindowExpression => return true
```

Comment:
we have couple other things in the analyzer. we should fix those in a separate pr.